packages/video_player/video_player_avfoundation/example/ios/RunnerUITests/VideoPlayerUITests.m

```diff
@@ -98,17 +99,22 @@ - (void)testEncryptedVideoStream {
     // 0.5 compression is good enough for debugging purpose.
     NSData *imageData = UIImageJPEGRepresentation(smallerImage, 0.5);
     NSString *imageString = [imageData base64EncodedStringWithOptions:0];
-    NSLog(@"frame %d image data:\n%@", i, imageString);
 
-    [frames addObject:imageString];
+    [distinctFrames addObject:imageString];
+    [allFrames addObject:imageString];
 
     // The sample interval must NOT be the same as video length.
     // Otherwise it would always result in the same frame.
     [NSThread sleepForTimeInterval:1];
   }
 
   // At least 1 loading and 2 distinct frames (3 in total) to validate that the video is playing.
-  XCTAssert(frames.count >= 3, @"Must have at least 3 distinct frames.");
+  if (distinctFrames.count < 3) {
+    for (int i = 0; i < allFrames.count; i++) {
```

Comment:
Small nit, you can use `enumerateObjectsUsingBlock:` if you want to fast enumerate and also need the index. https://developer.apple.com/documentation/foundation/nsarray/1415846-enumerateobjectsusingblock